File: VoteApp/app/src/main/java/com/example/voteapp/MainActivity.java
package com.example.voteapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setTitle("명화 선호도 투표");

        final int VoteCount[] = new int[9];
        for (int i = 0; i <VoteCount.length ; i++) {
            VoteCount[i]=0;
        }
        ImageView image[]= new ImageView[9];
        Integer imageID[]= {R.id.iv1,R.id.iv2,R.id.iv3,R.id.iv4,R.id.iv5,R.id.iv6,R.id.iv7,R.id.iv8,R.id.iv9};
        final String imageName[] = {"독서하는 소녀","꽃장식 모자 소녀","부채를 든 소녀","이레느깡 단 베르망","잠자는 소녀" ,
                "테라스의 두 자매","피아노 레슨","피아노 앞의 소녀들","해변에서"};
        for (int i = 0; i <imageID.length ; i++) {
            final int index;
            index = i;

            image[index]=(ImageView)findViewById(imageID[index]);
            image[index].setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    VoteCount[index]++;
                    Toast.makeText(MainActivity.this,imageName[index]+": 총"+VoteCount[index]+"표",Toast.LENGTH_SHORT).show();
                }
            });
        }

        Button btnResult =(Button)findViewById(R.id.btn_result);
        btnResult.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this,ResultActivity.class);
                intent.putExtra("VoteCount",VoteCount);
                intent.putExtra("ImageNames",imageName);
                startActivity(intent);
            }
        });
    }
}

File: Kotlin/SimpleDiary/app/src/main/java/com/example/simplediary/MainActivity.kt
package com.example.simplediary

import android.content.Context
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.NumberPicker
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.widget.AppCompatButton
import androidx.core.content.edit

class MainActivity : AppCompatActivity() {

    private val numPicker1 : NumberPicker by lazy {
        findViewById<NumberPicker>(R.id.np1)
            .apply{
                minValue = 0
                maxValue = 9
            }
    }

    private val numPicker2 : NumberPicker by lazy {
        findViewById<NumberPicker>(R.id.np2)
            .apply{
                minValue = 0
                maxValue = 9
            }
    }
    private val numPicker3 : NumberPicker by lazy {
        findViewById<NumberPicker>(R.id.np3)
            .apply{
                minValue = 0
                maxValue = 9
            }
    }

    private val openButton : AppCompatButton by lazy {
        findViewById(R.id.btn_open)
    }

    private val chageButton : AppCompatButton by lazy {
        findViewById(R.id.btn_change)
    }

    private var chagePasswordMode = false


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        numPicker1
        numPicker2
        numPicker3

        openButton.setOnClickListener {
            if(chagePasswordMode){
                Toast.makeText(this, "비밀번호가 변경중입니다.", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            val passwordPreFerences = getSharedPreferences("password", Context.MODE_PRIVATE)
            val userPassWord = "${numPicker1.value}${numPicker2.value}${numPicker3.value}" // 사용자가 입력한 비밀번호
            if(passwordPreFerences.getString("password", "000").equals(userPassWord)){
                // 초기값은 000으로 비밀번호 설정

                //votmdnjem tj
            }else{
                showErrorDialog()
            }
        }

        chageButton.setOnClickListener {
            val passwordPreFerences = getSharedPreferences("password", Context.MODE_PRIVATE)
            val userPassWord = "${numPicker1.value}${numPicker2.value}${numPicker3.value}" // 사용자가 입력한 비밀번호
            if(chagePasswordMode){


                passwordPreFerences.edit(true){
                    putString("password", userPassWord)
                }
                chagePasswordMode = false
                chageButton.setBackgroundColor(Color.BLACK)

            }else {
                if(passwordPreFerences.getString("password", "000").equals(userPassWord)){
                    chagePasswordMode = true
                    Toast.makeText(this, "변경할 패스워드를 입력해주세요.", Toast.LENGTH_SHORT).show()
                    chageButton.setBackgroundColor(Color.RED)
                }else{
                    showErrorDialog()
                }
            }
        }

    }

    private fun showErrorDialog(){
        AlertDialog.Builder(this)
            .setTitle("실패!!")
            .setMessage("비밀번호가 틀렸습니다.")
            .setPositiveButton("확인"){ _, _ -> }
            .create()
            .show()
    }
}